crates/server/src/rpc/warp_ws_transport.rs

```diff
@@ -23,62 +23,84 @@ impl WarpWebSocketTransport {
         Self {
             read: Mutex::new(read),
             write: Mutex::new(write),
-            ready: AtomicBool::new(false),
         }
     }
 }
 
 #[async_trait::async_trait]
 impl Transport for WarpWebSocketTransport {
-    async fn receive(&self) -> Result<TransportEvent, TransportError> {
+    async fn receive(&self) -> Result<TransportMessage, TransportError> {
         match self.read.lock().await.next().await {
             Some(Ok(message)) => {
                 if message.is_binary() {
-                    let message = self.message_to_transport_event(message.into_bytes());
-                    if let TransportEvent::Connect = message {
-                        self.ready.store(true, Ordering::SeqCst);
-                    }
-                    return Ok(message);
+                    let message_data = message.into_bytes();
+                    return Ok(message_data);
                 } else {
+                    if message.is_close() {
+                        return Err(TransportError::Closed);
+                    }
                     // Ignore messages that are not binary
-                    return Err(TransportError::Internal);
+                    error!("> WebSocketTransport > Received message is not binary");
+                    return Err(TransportError::NotBinaryMessage);
                 }
             }
             Some(Err(err)) => {
-                println!("Failed to receive message {err:?}");
+                error!(
+                    "> WebSocketTransport > Failed to receive message {}",
+                    err.to_string()
+                );
+                if let Some(transport_error) = translate_warp_error(&err) {
+                    Err(transport_error)
+                } else {
+                    Err(TransportError::Internal(Box::new(err)))
+                }
             }
             None => {
-                println!("No message")
+                error!("> WebSocketTransport > None received > Closing...");
+                return Err(TransportError::Closed);
             }
         }
-        println!("Closing transport...");
-        self.close().await;
-        Ok(TransportEvent::Close)
     }
 
     async fn send(&self, message: Vec<u8>) -> Result<(), TransportError> {
         let message = WarpWSMessage::binary(message);
-        self.write
-            .lock()
-            .await
-            .send(message)
-            .await
-            .map_err(|_| TransportError::Internal)?;
-        Ok(())
+        match self.write.lock().await.send(message).await {
+            Err(err) => {
+                error!(
+                    "> WebSocketTransport > Error on sending in a ws connection {}",
+                    err.to_string()
+                );
+                if let Some(transport_error) = translate_warp_error(&err) {
+                    Err(transport_error)
+                } else {
+                    Err(TransportError::Internal(Box::new(err)))
+                }
```

Comment:
Suggestion: This block is repeated a couple of times, should we think of extracting a function for it?